examples/terraform/openstack/variables.tf

```diff
@@ -30,47 +31,82 @@ variable "apiserver_alternative_names" {
   type        = list(string)
 }
 
-variable "worker_os" {
-  description = "OS to run on worker machines"
+variable "image" {
+  description = "image name to use"
+  default     = ""
+  type        = string
+}
 
-  # valid choices are:
-  # * ubuntu
-  # * centos
-  # * rockylinux
-  default = "ubuntu"
-  type    = string
+variable "image_properties_query" {
+  description = "in absence of var.image, this will be used to query API for the image"
+  default = {
+    os_distro  = "ubuntu"
+    os_version = "22.04"
+  }
+  type = map(any)
 }
 
-variable "ssh_public_key_file" {
-  description = "SSH public key file"
-  default     = "~/.ssh/id_rsa.pub"
+# Network Variables
+variable "subnet_cidr" {
+  default     = "192.168.1.0/24"
+  description = "OpenStack subnet cidr"
   type        = string
+
+  validation {
+    condition     = can(cidrhost(var.subnet_cidr, 32))
+    error_message = "Must be valid IPv4 CIDR."
+  }
 }
 
-variable "ssh_port" {
-  description = "SSH port to be used to provision instances"
-  default     = 22
+variable "external_network_name" {
+  description = "OpenStack external network name"
+  type        = string
+}
+
+variable "subnet_dns_servers" {
+  type    = list(string)
+  default = ["8.8.8.8", "8.8.4.4"]
+
+  validation {
+    condition = alltrue([
+      for a in var.subnet_dns_servers : can(regex("^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$", a))
+    ])
+    error_message = "All elements must be a valid IPv4 address."
+  }
+}
+
+# Controlplane Variables
+variable "control_plane_vm_count" {
+  description = "number of control plane instances"
+  default     = 3
   type        = number
 }
 
-variable "ssh_username" {
-  description = "SSH user, used only in output"
-  default     = "ubuntu"
+variable "control_plane_flavor" {
+  description = "OpenStack instance flavor for the control plane nodes"
+  default     = "m1.small"
   type        = string
 }
 
-variable "ssh_private_key_file" {
-  description = "SSH private key file used to access instances"
-  default     = ""
+# Worker Node Variables
+variable "worker_os" {
+  description = "OS to run on worker machines"
+  default     = "ubuntu"
   type        = string
+
+  validation {
+    condition     = contains(["ubuntu", "centos", "rockylinux"], var.worker_os)
```

Comment:
Sorry for misleading message, here we actually have more options. All that are supported by the machine-controller/OSM. And we even use them in out e2e tests. This strict validation will deny valid cases like `flatcar`, `rhel` or maybe any future OSes. I'd say that it's better to remove the validation at all.